src/maptm/src/maptm_ovsdb.c

```diff
@@ -0,0 +1,307 @@
+*
+* 2. Redistributions in binary form must reproduce the above copyright notice,
+*    this list of conditions and the following disclaimer in the documentation
+*    and/or other materials provided with the distribution.
+*
+* 3. Neither the name of the copyright holder nor the names of its contributors
+*    may be used to endorse or promote products derived from this software
+*    without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+* AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+* IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+* ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+* LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+* CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+* SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+* INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+* CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#include <stdio.h>
+
+#include "ovsdb_update.h"
+#include "ovsdb_sync.h"
+#include "ovsdb_table.h"
+#include "schema.h"
+
+#include "maptm.h"
+
+#ifdef MAPTM_DEBUG
+#undef LOGI
+#define LOGI    printf
+#endif
+
+/**
+ * Globals
+ */
+
+/* Log entries from this file will contain "OVSDB" */
+#define MODULE_ID LOG_MODULE_ID_OVSDB
+#define MAPTM_IFC_WAN "br-wan"
+
+ovsdb_table_t table_Netfilter;
+ovsdb_table_t table_Node_Config;
+struct ovsdb_table table_Node_State;
+static ovsdb_table_t table_Interface;
+static ovsdb_table_t  table_IP_Interface;
+ovsdb_table_t  table_Wifi_Inet_Config;
+ovsdb_table_t  table_DHCPv6_Client;
+ovsdb_table_t  table_DHCP_Client;
+ovsdb_table_t  table_IPv6_Address;
+
+/* To change to Enum  */
+int WanConfig = 0;
+
+// Testing MAP-T tables presence
+bool maptm_ovsdb_tabs_ready()
+{
+    json_t *where_config = NULL;
+    bool retval = false;
+    
+    //Checking MAP-T entry presence in Node_Config table
+    where_config = ovsdb_where_multi(
+            ovsdb_where_simple_typed(SCHEMA_COLUMN(Node_Config, module),"WANO", OCLM_STR),
+            ovsdb_where_simple_typed(SCHEMA_COLUMN(Node_Config, key),"maptParams", OCLM_STR),
+            NULL);
+    if(!where_config)
+    {
+        LOGI("Node_Config table is not set by the cloud yet!");
+        retval = false; // switch to Cloud independant mode
+    }
+    
+    return retval;
+}
+
+// Make MAPT_Support parameter persistent
+bool maptm_persistent(void)
+{
+    bool ret = false;
+    char mapt_support[10];
+    osp_ps_t *ps = NULL;
+
+    ps = osp_ps_open("MAPT_SUPPORT", OSP_PS_RDWR | OSP_PS_PRESERVE);
+    if (ps == NULL)
+    {
+        LOG(ERR, "maptm: Error opening \"%s\" persistent store.", mapt_support);
```

Comment:
```suggestion
        LOGE("maptm: Error opening persistent store \"%s\"", MAPT_SUPPORT_STORE);
```
`mapt_support` is uninitialized at this point.
String "MAPT_SUPPORT" is used multiple times, define it, e.g."
```
#define MAPT_SUPPORT_STORE "MAPT_SUPPORT"
```